### PR TITLE
Disable running windows tests unless Run Windows lebel set

### PR DESCRIPTION
**Pull request by bogdandrutu**
Tests will still be run on main, so we will not release without them passing, but we don't run on each PR unless approver/maintainer believes they are needed (windows specific code is changed) then use "Run Windows" label

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]
I like the compromise of being able to trigger them manually by adding a label.

**Review by evan-bradley — APPROVED** [review]

File: .github/workflows/build-and-test-windows.yml [hunk 1]
```diff
@@ -28,7 +31,7 @@ jobs:
           - internal
           - other
     runs-on: windows-latest
-    if: ${{ github.actor != 'dependabot[bot]' }}
+    if: ${{ github.actor != 'dependabot[bot]' && contains(github.event.pull_request.labels.*.name, 'Run Windows') }}
```
Comment (TylerHelmuth) [inline .github/workflows/build-and-test-windows.yml:34]: Will this block execution on when the PR is merged into main if the PR did not have the `Run Windows` label?

**Review by TylerHelmuth — COMMENTED** [review]
I love the idea of speeding up our CI for a majority of PRs.  My only concern is how will we know in a timely manner when a PR that didn't run windows tests actually breaks these tests?  

The CI fails on main sometimes and I get the sense that we aren't checking that regularly (I definitely am not).  How will we differentiate between a main CI failure that is a windows test vs a flaky test or something else?  I don't really want to be checking the Actions tab regularly. 

Worse case scenario would be that we get to release day, add the `Run Windows` label to the prep PR and realize that those main CI failures were real and now the release is blocked bc we gotta mess with a PR that broke windows builds.

Another situation that we might get into is a PR sneaks by the CI, starts breaking builds on main, and then another PR is opened that needs the `Run Windows` label and now it isn't clear if the windows builds are failing bc of the previous PR or bc of this labeled PR.  That would certainly be a negative experience for new contributors.

File: .github/workflows/build-and-test-windows.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/workflows/build-and-test-windows.yml:34]: Good point, need to fix that

**Comment by bogdandrutu** [comment]
@TylerHelmuth I understand your concern, but this repo is "unmaintainable" as it is right now. To merge a PR you need to wait 1h, that is against any productivity rule ...

Give me other options and happy to consider them.

**Comment by TylerHelmuth** [comment]
> To merge a PR you need to wait 1h, that is against any productivity rule

Yes this is certainly unacceptable. 

Disabling flaky tests will go a long way towards making the build more reliable.  That's probably the solution to my concern.  I don't have an alternative solution in mind, just wanted to put the thoughts out there for people to mull over.

**Comment by dmitryax** [comment]
I'm good with this PR, but we have a few components (at least 3 receivers) that are Windows only. Maybe we should always run tests that have only windows build flag? That shouldn't be a lot of overhead and we won't lose test coverage for each PR

**Comment by bogdandrutu** [comment]
Not sure how "Maybe we should always run tests that have only windows build flag?". I thought about this.

**Comment by TylerHelmuth** [comment]
@evan-bradley could the add-labels workflow be updated to automatically add the 'Run Windows' label for specific components?

**Comment by dmitryax** [comment]
I believe `go test --tags=windows ./...` should work. we just need to create a new make target

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Created an issue to follow up https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16463

**Pull request merged**